common/nginx/etc/nginx.conf

```diff
@@ -19,13 +19,13 @@ http {
                 }
 
                 location / {
-                        proxy_set_header Host $host:$server_port;
+                        proxy_set_header Host $http_host;
```

Comment:
one subtle aspect of `$http_host` if no `Host` http header present on request, is still valid http, but this variable will be blank, which may result in downstream services responses like:

```
 "_links": {
        "account": {
            "href": "http:///accounts/{account_id}",
            "templated": true
        },
```
it might be possible to dynamically switch to `$host` in that case which pulls the host name from the request line instead, I haven't tried this yet, but looking at examples of using if statements like:

```
if ( $http_host = '' ) {
        // no http Host header was sent
        // maybe just error out requests that don't provide host header?
        return 444;
        // or use host derived from request line instead
        proxy_set_header Host $host;
} else {
       // use the http Host header
       proxy_set_header Host $http_host;
}

```

Reply:
This case is already handled I believe by nginx. For example:
```
❯ curl -H 'Host:' http://localhost:8000
<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx/1.18.0 (Ubuntu)</center>
</body>
</html>
```

Have you seen otherwise?

Reply:
@leighmcculloch , nice, yes, that example was what I had wondered about, I tried same locally.and got 400 error also when host header is absent. Now I know! default nginx will reject requests with missing host headers, thanks for checking!